README.md

```diff
@@ -1,34 +1,87 @@
 # xtp-bindgen
 
-XTP Bindgen is an open source framework to generate PDK bindings for [Extism](https://extism.org) plug-ins.
-It's used by the [XTP Platform](https://www.getxtp.com/), but can be used outside of the platform to define
-any Extism compatible plug-in system.
+XTP Bindgen is an open source framework to generate PDK bindings for
+[Extism](https://extism.org) plug-ins. It's used by the
+[XTP Platform](https://www.getxtp.com/), but can be used outside of the platform
+to define any Extism compatible plug-in system.
+
+This repository hosts the core schema parser and validation code, and is not
+meant for typical day-to-day use. Instead, you should use the `xtp` CLI to
+generate bindings from an
+[XTP Schema](https://docs.xtp.dylibso.com/docs/concepts/xtp-schema).
+
+## Quickstart
+
+### 1. Install the `xtp` CLI.
+
+See installation instructions
+[here](https://docs.xtp.dylibso.com/docs/cli#installation).
+
+### 2. Create a schema using our OpenAPI-inspired IDL:
+
+```yaml
+version: v1-draft
+exports: 
+  CountVowels:
+      input: 
+          $ref: "#/components/schemas/Vowels"
+          contentType: application/json
+      output:
+          $ref: "#/components/schemas/Count"
+          contentType: application/json
```

Comment:
Should we make this something like?

```
version: v1-draft
exports: 
  countVowels:
      input: 
          type: string
          contentType: plaint/text; charset=utf-8
      output:
          $ref: "#/components/schemas/VowelReport"
          contentType: application/json
```

Also, It might be confusing if we don't provide where the types are coming from. Linking to example schema works maybe just a yaml comment here about "component section here" or something

Reply:
Yes I updated this and will use the same example in the other extism pdk repos!